### PR TITLE
Fixed board selection not sticking

### PR DESCRIPTION
**Pull request by blckmn**
What it says on the tin. The board selection was not sticking.

### CODE REVIEW

**Comment by haslinghuis** [comment]
This does not expose the cloud build options anymore.

**Review by haslinghuis — CHANGES_REQUESTED** [review]
This does not expose the cloud build options anymore.

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> PASS
 - cooling off period lapsed -> FAIL
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> FAIL
 - `Tested` label found -> FAIL
 - assigned to an approver -> PASS
 - approver count at least three -> FAIL

**Comment by blckmn** [comment]
> This does not expose the cloud build options anymore.

There was a small bug where call trigger change didn't actually work if it didn't change. Fixed by calling directly.

File: src/js/tabs/firmware_flasher.js
Comment (haslinghuis) [inline src/js/tabs/firmware_flasher.js:7]: selectedBoard is used in protocols/stm32.js for validation on flashing a different target.

Comment (blckmn) [reply src/js/tabs/firmware_flasher.js]: Reverted :) That dependency should probably be removed.

**Review by haslinghuis — APPROVED** [review]

**Pull request merged**
